apollo-router/src/warp_http_server_factory.rs

```diff
@@ -470,17 +544,26 @@ where
                     .map(|response| {
                         tracing::trace_span!("serialize_response")
                             .in_scope(|| {
-                                response.map(|body| {
-                                    Bytes::from(
-                                        serde_json::to_vec(&body)
+                                response.map(|body| match body {
+                                    ResponseBody::GraphQL(res) => Bytes::from(
+                                        serde_json::to_vec(&res)
                                             .expect("responsebody is serializable; qed"),
-                                    )
+                                    ),
+                                    ResponseBody::RawJSON(res) => Bytes::from(
+                                        serde_json::to_vec(&res)
+                                            .expect("responsebody is serializable; qed"),
+                                    ),
+                                    ResponseBody::RawString(res) => Bytes::from(
+                                        serde_json::to_vec(&res)
+                                            .expect("responsebody is serializable; qed"),
+                                    ),
+                                    ResponseBody::Text(res) => Bytes::from(res),
```

Comment:
NIT: Could this be collapsed into:
```
ResponseBody::GraphQL(res) | ResponseBody::RawJSON(res) | ResponseBody::RawString(res) => Bytes::from(
                                        serde_json::to_vec(&res)
                                            .expect("responsebody is serializable; qed"),
                                    ),
```

Not sure this is much neater, but ...

Reply:
No it won't work because the `res` value has not the same type